source/extensions/filters/network/rbac/rbac_filter.cc

```diff
@@ -39,26 +39,33 @@ Network::FilterStatus RoleBasedAccessControlFilter::onData(Buffer::Instance&, bo
                 : "none",
             callbacks_->connection().streamInfo().dynamicMetadata().DebugString());
 
+  std::string log_policy_id = "none";
   // When the enforcement type is continuous always do the RBAC checks. If it is a one time check,
   // run the check once and skip it for subsequent onData calls.
   if (config_->enforcementType() ==
       envoy::extensions::filters::network::rbac::v3::RBAC::CONTINUOUS) {
-    shadow_engine_result_ = checkEngine(Filters::Common::RBAC::EnforcementMode::Shadow);
-    engine_result_ = checkEngine(Filters::Common::RBAC::EnforcementMode::Enforced);
+    shadow_engine_result_ = checkEngine(Filters::Common::RBAC::EnforcementMode::Shadow).first;
+    auto result = checkEngine(Filters::Common::RBAC::EnforcementMode::Enforced);
+    engine_result_ = result.first;
+    log_policy_id = result.second;
   } else {
     if (shadow_engine_result_ == Unknown) {
       // TODO(quanlin): Pass the shadow engine results to other filters.
-      shadow_engine_result_ = checkEngine(Filters::Common::RBAC::EnforcementMode::Shadow);
+      shadow_engine_result_ = checkEngine(Filters::Common::RBAC::EnforcementMode::Shadow).first;
     }
 
     if (engine_result_ == Unknown) {
-      engine_result_ = checkEngine(Filters::Common::RBAC::EnforcementMode::Enforced);
+      auto result = checkEngine(Filters::Common::RBAC::EnforcementMode::Enforced);
+      engine_result_ = result.first;
+      log_policy_id = result.second;
     }
   }
 
   if (engine_result_ == Allow) {
     return Network::FilterStatus::Continue;
   } else if (engine_result_ == Deny) {
+    callbacks_->connection().streamInfo().setResponseCodeDetails(
```

Comment:
Sorry I missed this earlier (I was focused on the filters/http part) but I don't think setting responseCodeDetails on the network filter makes sense per se.  Response code details is an HTTP / L7 construct.  Unfortunately I don't think there is a network (L4) equivalent for downstream.  It might make more sense to add one than reuse the HTTP field.

Reply:
okay I reverted the change in the network filter, I feel we really should have a network equivalent, or maybe have a general `RESPONSE_DETAILS` so that it can be used in both L4 and L7?